xugrid/regrid/regridder.py

```diff
@@ -155,17 +165,16 @@ def _regrid_array(self, source):
                 chunks=chunks,
                 meta=np.array((), dtype=source.dtype),
             )
+            out = out.compute().reshape(out_shape)
         elif isinstance(source, np.ndarray):
             out = self._regrid(source, self._weights, size)
+            out = out.reshape(out_shape)
         else:
             raise TypeError(
                 "Expected dask.array.Array or numpy.ndarray. Received: "
                 f"{type(source).__name__}"
             )
-
-        # E.g.: sizes of ("time", "layer") + ("y", "x")
-        out_shape = first_dims_shape + self._target.shape
-        return out.reshape(out_shape)
```

Comment:
Because the reshape does not work for a dask-array. For the dask-array I first used the dask.array.reshape(), which gives the result that we want. For the current implementation (see reply above) I could move it back indeed.

Reply:
Ah an understandable struggle with dask.array.reshape :-) I would prefer if you moved it back down as because of the call to ``dask.array.compute`` we can leave the code as it was.